### PR TITLE
Governable parameters improvements

### PR DESCRIPTION
**Pull request by pdyraga**
List of major and minor tweaks after the review of #2613. I recommend reviewing this PR commit by commit. The most important changes are:
- Improved documentation of `RandomBeacon` contract public fields, did small documentation tweaks for `RandomBeaconGovernance`.
- Extracted governance delays to constants and explained them.
- Updated order of tests in `RandomBeaconGovernance.test.ts` to reflect the order of functions in `RandomBeaconGovernance` contract.
- Proposed initial parameter values for `RandomBeacon` contract. The initial parameters assigned in the constructor ensures the beacon is able to work safely but they are just proposed defaults and they might be updated with `update*` function call after `RandomBeacon` contract deployment and before transferring the ownership to the `RandomBeaconGovernance` contract.
- Group creation frequency made a standard parameter. Group creation frequency is strongly correlated with relay request fee and they should both be updated at the same time. Moreover, even if the group creation will be triggered too often, honest stakers may decide to opt-out and do not participate in DKG. They will lose future rewards but they will not have their stake slashed. This is the same situation as if they undelegated and exited the sortition pool.
- Reduced standard parameter governance delay to 12h. 12h is enough, none of these parameters affect stakes of honest stakers.
- Improved allowed governable parameter ranges. Improved revert messages and parameter ranges to be inclusive for values where applicable. Improved test to better cover corner cases. The current version is:
  - Relay entry submission eligibility delay must be > 0
  - Callback gas limit must be > 0 and <= 1000000
  - Group creation frequency must be > 0
  - Group lifetime must be >= 1 day and <= 2 weeks
  - DKG result challenge period length must be >= 10
  - DKG result submission eligibility delay must be > 0

### CODE REVIEW

File: solidity/random-beacon/contracts/RandomBeacon.sol [hunk 1]
```diff
@@ -16,47 +16,116 @@ pragma solidity ^0.8.6;
...
+    /// @notice Reward in T for unlocking the sortition pool if DKG timed out.
+    ///         When DKG result submission timed out, sortition pool is still
+    ///         locked and someone needs to unlock it. Anyone can do it and earn
+    ///         `sortitionPoolUnlockingReward`.
     uint256 public sortitionPoolUnlockingReward;
 
-    /// @notice Slashing amount for not submitting relay entry
+    /// @notice Slashing amount for not submitting relay entry. When 
+    ///         relay entry hard timeout is reached without the relay entry
+    ///         submitted, each group members gets slashed for
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeacon.sol:115]: members => member

File: solidity/random-beacon/contracts/RandomBeacon.sol [hunk 2]
```diff
@@ -16,47 +16,116 @@ pragma solidity ^0.8.6;
...
+    ///         `relayEntrySubmissionFailureSlashingAmount`. If the relay entry
+    ///         gets submitted after the soft timeout (see 
+    ///         `relayEntrySubmissionEligibilityDelay` documentation), but
+    ///         before the hard timeout, each group members gets slashed
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeacon.sol:119]: members => member

File: solidity/random-beacon/test/RandomBeaconGovernance.test.ts [hunk 3]
```diff
@@ -18,6 +31,27 @@ describe("RandomBeaconGovernance", () => {
...
+      initialGroupCreationFrequency,
+      initialGroupLifeTime,
+      initialDkgResultChallengePeriodLength,
+      initialDkgResultSubmissionEligibilityDelay
+    )
+    await randomBeacon.connect(governance).updateRewardParameters(
+      initialDkgResultSubmissionReward,
+      initialSortitionPoolUnlockingReward
+    )
+    await randomBeacon.updateSlashingParameters(
```
Comment (tomaszslabon) [inline solidity/random-beacon/test/RandomBeaconGovernance.test.ts:50]: `await randomBeacon.updateSlashingParameters` => `await randomBeacon.connect(governance).updateSlashingParameters`, just like in the above `update*` calls?

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol [hunk 4]
```diff
@@ -17,9 +17,10 @@ pragma solidity ^0.8.6;
 import "./RandomBeacon.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 
-/// @title Random beacon governance
-/// @notice Random beacon governance owns the random beacon contract and is
-///         responsible for updating its governable parameters
+/// @title Keep Random Beacon Governance
+/// @notice Owns the `RandomBeacon` contract and is responsible for updating its
+///         governable parameters in respect to governance delay invividual
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeaconGovernance.sol:22]: invividual => individual

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol [hunk 5]
```diff
@@ -59,6 +60,32 @@ contract RandomBeaconGovernance is Ownable {
...
+    // The full list of parameters protected by this delay:
+    // - relay entry hard timeout
+    // - callback gas limit
+    // - group creation frequency
+    // - group lifetime
+    // - relay entry submission failure slashing amount
+    uint256 internal CRITICAL_PARAMETER_GOVERNANCE_DELAY = 2 weeks;
+
+    // Short governance delay for non-critical parameters. Honest stakers should
+    // not be severly affected by any change of these parameters.
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeaconGovernance.sol:77]: severly => severely

File: solidity/random-beacon/contracts/RandomBeacon.sol [hunk 6]
```diff
@@ -109,36 +180,38 @@ contract RandomBeacon is Ownable {
...
-    ///      random beacon governance contract
+    ///      random beacon governance contract. The caller is responsible for
+    ///      validating parameters.
     /// @param _groupCreationFrequency New group creation frequency
     /// @param _groupLifetime New group lifetime
     /// @param _dkgResultChallengePeriodLength New DKG result challenge period
     ///        length
-    /// @param _dkgSubmissionEligibilityDelay New DKG submission eligibility
-    ///        delay
+    /// @param _dkgResultSubmissionEligibilityDelay New DKG submission
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeacon.sol:191]: "New DKG submission" => "New DKG result submission"

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol [hunk 7]
```diff
@@ -461,20 +502,20 @@ contract RandomBeaconGovernance is Ownable {
 
     /// @notice Begins the DKG submission eligibility delay update process.
     /// @dev Can be called only by the contract owner.
-    /// @param _newDkgSubmissionEligibilityDelay New DKG submission eligibility
-    ///        delay in blocks
-    function beginDkgSubmissionEligibilityDelayUpdate(
-        uint256 _newDkgSubmissionEligibilityDelay
+    /// @param _newDkgResultSubmissionEligibilityDelay New DKG submission
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeaconGovernance.sol:505]: "New DKG submission" => "New DKG result submission"

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol [hunk 8]
```diff
@@ -461,20 +502,20 @@ contract RandomBeaconGovernance is Ownable {
...
+    /// @param _newDkgResultSubmissionEligibilityDelay New DKG submission 
+    ///        eligibility delay in blocks
+    function beginDkgResultSubmissionEligibilityDelayUpdate(
+        uint256 _newDkgResultSubmissionEligibilityDelay
     ) external onlyOwner {
         /* solhint-disable not-rely-on-time */
         require(
-            _newDkgSubmissionEligibilityDelay > 0,
+            _newDkgResultSubmissionEligibilityDelay > 0,
             "DKG submission eligibility delay must be greater than 0 blocks"
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeaconGovernance.sol:513]: "DKG submission" => "DKG result submission"

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol [hunk 9]
```diff
@@ -461,20 +502,20 @@ contract RandomBeaconGovernance is Ownable {
 
     /// @notice Begins the DKG submission eligibility delay update process.
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeaconGovernance.sol:503]: "Begins the DKG submission" => "Begins the DKG result submission"

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol [hunk 10]
```diff
@@ -483,25 +524,25 @@ contract RandomBeaconGovernance is Ownable {
     /// @notice Finalizes the DKG submission eligibility delay update process.
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeaconGovernance.sol:524]: "Finalizes the DKG submission" => "Finalizes the DKG result submission"

File: solidity/random-beacon/contracts/RandomBeaconGovernance.sol [hunk 11]
```diff
@@ -760,22 +801,22 @@ contract RandomBeaconGovernance is Ownable {
         return
             getRemainingChangeTime(
                 dkgResultChallengePeriodLengthChangeInitiated,
-                24 hours
+                STANDARD_PARAMETER_GOVERNANCE_DELAY
             );
     }
 
     /// @notice Get the time remaining until the DKG submission eligibility
```
Comment (tomaszslabon) [inline solidity/random-beacon/contracts/RandomBeaconGovernance.sol:808]: "Get the time remaining until the DKG submission eligibility" => " Get the time remaining until the DKG result submission eligibility"

File: solidity/random-beacon/test/RandomBeacon.test.ts [hunk 12]
```diff
@@ -137,8 +139,8 @@ describe("RandomBeacon", () => {
       })
 
       it("should update the DKG submission eligibility delay", async () => {
```
Comment (tomaszslabon) [inline solidity/random-beacon/test/RandomBeacon.test.ts:141]: "should update the DKG submission eligibility delay" => "should update the DKG result submission eligibility delay"

**Comment by pdyraga** [comment]
All comments addressed, PR undrafted, changes from `main` reverse-merged, and I plan no more changes here. This is ready for a review!

File: solidity/random-beacon/contracts/RandomBeacon.sol (hunk 6 above)
Comment (tomaszslabon) [reply solidity/random-beacon/contracts/RandomBeacon.sol:210]: I think this one is still missing.

Comment (pdyraga) [reply solidity/random-beacon/contracts/RandomBeacon.sol]: Ops, my bad. Please see https://github.com/keep-network/keep-core/pull/2638/commits/bce451f4adca8b55455640f93d918679a107d783.

**Review by tomaszslabon — APPROVED** [review]
